pkgs/development/libraries/rocfft/default.nix

```diff
@@ -16,94 +18,140 @@
-  ] ++ lib.optionals (buildTests || buildBenchmarks) [
-    boost
-    fftw
-    fftwFloat
-    openmp
-  ];
-
-  propogatedBuildInputs = lib.optionals buildTests [
-    fftw
-    fftwFloat
-  ];
-
-  cmakeFlags = [
-    "-DCMAKE_C_COMPILER=hipcc"
-    "-DCMAKE_CXX_COMPILER=hipcc"
-    "-DUSE_HIP_CLANG=ON"
-    "-DSQLITE_USE_SYSTEM_PACKAGE=ON"
-    # Manually define CMAKE_INSTALL_<DIR>
-    # See: https://github.com/NixOS/nixpkgs/pull/197838
-    "-DCMAKE_INSTALL_BINDIR=bin"
-    "-DCMAKE_INSTALL_LIBDIR=lib"
-    "-DCMAKE_INSTALL_INCLUDEDIR=include"
-  ] ++ lib.optionals buildTests [
-    "-DBUILD_CLIENTS_TESTS=ON"
-  ] ++ lib.optionals buildBenchmarks [
-    "-DBUILD_CLIENTS_RIDER=ON"
-    "-DBUILD_CLIENTS_SAMPLES=ON"
-  ];
-
-  postInstall = lib.optionalString buildTests ''
-    mkdir -p $test/{bin,lib/fftw}
-    cp -a $out/bin/* $test/bin
-    ln -s ${fftw}/lib/libfftw*.so $test/lib/fftw
-    ln -s ${fftwFloat}/lib/libfftw*.so $test/lib/fftw
-    rm -r $out/lib/fftw
-    rm $test/bin/{rocfft_rtc_helper,*-rider} || true
-  '' + lib.optionalString buildBenchmarks ''
-    mkdir -p $benchmark/bin
-    cp -a $out/bin/* $benchmark/bin
-    rm $benchmark/bin/{rocfft_rtc_helper,*-test} || true
-  '' + lib.optionalString (buildTests || buildBenchmarks ) ''
-    mv $out/bin/rocfft_rtc_helper $out
-    rm -r $out/bin/*
-    mv $out/rocfft_rtc_helper $out/bin
+let
+  rocfft = stdenv.mkDerivation (finalAttrs: {
+    pname = "rocfft";
+    version = "5.4.2";
+
+    outputs = [
+      "out"
+    ] ++ lib.optionals buildTests [
+      "test"
+    ] ++ lib.optionals buildBenchmarks [
+      "benchmark"
+    ];
+
+    src = fetchFromGitHub {
+      owner = "ROCmSoftwarePlatform";
+      repo = "rocFFT";
+      rev = "rocm-${finalAttrs.version}";
+      hash = "sha256-NsYeEoBQ/0z31ZQ32l7N+qavWEVkH37snkTHntot7nE=";
+    };
+
+    nativeBuildInputs = [
+      cmake
+      rocm-cmake
+      hip
+    ];
+
+    buildInputs = [
+      sqlite
+      python3
+    ] ++ lib.optionals buildTests [
+      gtest
+    ] ++ lib.optionals (buildTests || buildBenchmarks) [
+      rocrand
+      boost
+      fftw
+      fftwFloat
+      openmp
+    ];
+
+    propogatedBuildInputs = lib.optionals buildTests [
+      fftw
+      fftwFloat
+    ];
+
+    cmakeFlags = [
+      "-DCMAKE_C_COMPILER=hipcc"
+      "-DCMAKE_CXX_COMPILER=hipcc"
+      "-DUSE_HIP_CLANG=ON"
+      "-DSQLITE_USE_SYSTEM_PACKAGE=ON"
+      # Manually define CMAKE_INSTALL_<DIR>
+      # See: https://github.com/NixOS/nixpkgs/pull/197838
+      "-DCMAKE_INSTALL_BINDIR=bin"
+      "-DCMAKE_INSTALL_LIBDIR=lib"
+      "-DCMAKE_INSTALL_INCLUDEDIR=include"
+    ] ++ lib.optionals buildTests [
+      "-DBUILD_CLIENTS_TESTS=ON"
+    ] ++ lib.optionals buildBenchmarks [
+      "-DBUILD_CLIENTS_RIDER=ON"
+      "-DBUILD_CLIENTS_SAMPLES=ON"
+    ];
+
+    postInstall = lib.optionalString buildTests ''
+      mkdir -p $test/{bin,lib/fftw}
+      cp -a $out/bin/* $test/bin
+      ln -s ${fftw}/lib/libfftw*.so $test/lib/fftw
```

Comment:
replacing with `.so` helps with `stdenv.hostPlatform.extensions.sharedLibrary`  building on other platforms. If it's only supported on Linux then platforms should be set.

Reply:
Forgive me, but this is a bit vague.
Are you saying we should just set platforms to linux?
AFAIK this should already be covered under ``hip``.
See: https://github.com/NixOS/nixpkgs/blob/b974cf6522a8a94228a6019a12a4aa2fce239d62/pkgs/development/compilers/hip/default.nix

Reply:
If this is linux-only, hardcoding .so extension is ok, otherwise use the variable which will give the right one for the different platforms.

And platforms should be set for all packages preferably.

Reply:
Yeah, the matrix chat also clarified that for me.
Thank you!